dialect_mysql.go

```diff
@@ -72,24 +72,25 @@ func (d MySQLDialect) ToSqlType(val reflect.Type, maxsize int, isAutoIncr bool)
 		return "datetime"
 	}
 
-	if maxsize < 1 {
-		maxsize = 255
-	}
-
 	/* == About varchar(N) ==
 	 * N is number of characters.
-	 * A varchar column can store up to 65535 bytes.
-	 * Remember that 1 character is 3 bytes in utf-8 charset.
-	 * Also remember that each row can store up to 65535 bytes,
-	 * and you have some overheads, so it's not possible for a
-	 * varchar column to have 65535/3 characters really.
-	 * So it would be better to use 'text' type in stead of
-	 * large varchar type.
+	 * According to the documentation, 0 < N <= 65535.
+	 * But one utf-8 character takes 3 bytes and each row can
+	 * be only 65535 bytes. So there is no way to know exactly
+	 * how many chars a varchar column can actually store.
+	 * Text columns contribute only up to 12 bytes to the row
+	 * size as their contents are stored off-page.
+	 * Hence, we use a conservative maximum of 512 for varchar,
+	 * after which we switch to the text type.
 	 */
-	if maxsize < 256 {
+	if maxsize == 0 {
+		// Closer match for unbounded text
+		return "longtext"
```

Comment:
As per https://dev.mysql.com/doc/refman/8.0/en/string-type-overview.html, `longtest` is:
> A TEXT column with a maximum length of 4,294,967,295 or 4GB (232 − 1) characters.

In the discussion on the ticket, you mentioned:
> I want to follow @lieut-data suggestion but use "varchar(16384)" instead of "text" for maxsize = 0.

Can you elaborate on the thinking here? Isn't using `longtext` basically the same as `text`, which invokes the issues @cpoile mentioned on the ticket?

Reply:
Yes, the change on mind here is related to the comments in the PR #4, where is more "expected" by a user, if I don't give you any boundary in the field, I want an unbounded field (that is the behavior for postgres) and because it is about consistency, I think here it makes sense.

Reply:
Discussed offline with @jespino, but we agreed `longtext` makes sense as proposed, though we'll never plan to actually use it in mattermost-server, preferring instead to always set an explicit length.